grails-app/services/org/transmartproject/security/LoginAttemptCacheService.groovy

```diff
@@ -0,0 +1,94 @@
+package org.transmartproject.security
+
+import com.google.common.cache.Cache
+import com.google.common.cache.CacheBuilder
+import com.google.common.cache.CacheLoader
+
+import javax.annotation.PostConstruct
+import java.util.concurrent.TimeUnit
+
+/**
+ * This class counts failed login attempts per account.
+ * It has functionality for checking when given account should be locked and on how long.
+ *
+ * Original version by Grygoriy Mykhalyuno could be found by below link
+ * http://www.grygoriy.com/blog/2012/10/06/prevent-brute-force-attack-with-spring-security/
+ */
+class LoginAttemptCacheService {
+
+    private Cache failedAttempts
+
+    private int allowedNumberOfAttempts
+
+    int getAllowedNumberOfAttempts() { allowedNumberOfAttempts }
+
+    private int lockTimeInMinutes
+
+    int getLockTimeInMinutes() { lockTimeInMinutes }
+
+    def grailsApplication
+
+    final static int DEFAULT_ALLOWED_NUMBER_OF_ATTEMPTS = 3
+    final static int DEFAULT_LOGIN_ATTEMPTS_LOCK_MINUTES = 10
+
+    @PostConstruct
+    void init() {
+        def confAllowedNumberOfAttempts = grailsApplication.config.brutforce.loginAttempts.allowedNumberOfAttempts
+        if (!confAllowedNumberOfAttempts || confAllowedNumberOfAttempts < 0) {
+            log.warn("allowedNumberOfAttempts is expected to be greater than 0," +
+                    " but was ${confAllowedNumberOfAttempts}. " +
+                    "Fallback to default value ${DEFAULT_ALLOWED_NUMBER_OF_ATTEMPTS}")
+            allowedNumberOfAttempts = DEFAULT_ALLOWED_NUMBER_OF_ATTEMPTS
+        } else {
+            allowedNumberOfAttempts = confAllowedNumberOfAttempts
+            log.debug "set allowedNumberOfAttempts to ${allowedNumberOfAttempts}"
+        }
+
+        def confBlockMinutes = grailsApplication.config.brutforce.loginAttempts.lockTimeInMinutes
+        if (!confBlockMinutes || confBlockMinutes < 0) {
+            log.warn("lockTimeInMinutes is expected to be greater than 0," +
+                    " but was ${confBlockMinutes}. " +
+                    "Fallback to default value ${DEFAULT_LOGIN_ATTEMPTS_LOCK_MINUTES}")
+            lockTimeInMinutes = DEFAULT_LOGIN_ATTEMPTS_LOCK_MINUTES
+        } else {
+            lockTimeInMinutes = confBlockMinutes
+            log.debug "set lockTimeInMinutes to ${lockTimeInMinutes}"
+        }
+
+        failedAttempts = CacheBuilder.newBuilder()
+                .expireAfterWrite(lockTimeInMinutes, TimeUnit.MINUTES)
+                .build({ 0 } as CacheLoader)
```

Comment:
Several issues here. First it's arguably a separation of concerns to have a class take care of configuring itself. Second I see you here introducing new configuration settings that don't show up in `Config.groovy`. This means by default transmart will emit warnings. Third, if the configuration doesn't exist, this will emit rather strange messages as far as I can see. `grailsApplication.config.brutforce.loginAttempts.lockTimeInMinutes` will return an empty `ConfigObject`; when toString() is called you'll get sth like ConfigObject@XXXX.

Suggestion:
Move the default values to `Config.groovy` (I'd actually prefer this disabled by default). Inject the settings in the Spring configuration. On init(), only validate the settings and assign `failedAttempts`.